### PR TITLE
Adding labels from envelope fields

### PR DESCRIPTION
**Pull request by nrxus**
We also moved the dev functions into runnable go scripts (print firehose to stdout, clear SD metrics, send mock metric to SD every x seconds)

### CODE REVIEW

File: src/stackdriver-nozzle/nozzle/envelope_test.go [hunk 1]
```diff
@@ -0,0 +1,82 @@
...
+		}
+
+		labels := envelope.Labels()
+		Expect(labels).To(Equal(map[string]string{
+			"origin": origin,
+			"event_type": eventType.String(),
+			"deployment": deployment,
+			"job": job,
+			"index": index,
+			"ip": ip,
```
Comment (johnsonj) [inline src/stackdriver-nozzle/nozzle/envelope_test.go]: shouldn't there be a field here for tags? Or maybe you want to add an (Envelope*) func Payload() that adds the tags there?

File: src/stackdriver-nozzle/scripts/repro-metrics-rpc-errors.go [hunk 2]
```diff
@@ -0,0 +1,70 @@
+package main
```
Comment (johnsonj) [inline src/stackdriver-nozzle/scripts/repro-metrics-rpc-errors.go]: can you add some documentation to this one? eg, what's the error, why we have this code

File: src/stackdriver-nozzle/scripts/firehose-to-stdout.go [hunk 3]
```diff
@@ -0,0 +1,26 @@
+package main
```
Comment (johnsonj) [inline src/stackdriver-nozzle/scripts/firehose-to-stdout.go]: not a hard stance here but i think this one may be appropriate as a part of the main binary. That said we only expect these to get ran by bosh where someone won't use that output so feel free to ignore it.

File: src/stackdriver-nozzle/config/config.go [hunk 4]
```diff
@@ -0,0 +1,42 @@
+package config
```
Comment (johnsonj) [inline src/stackdriver-nozzle/config/config.go]: nice move on breaking these components out

File: src/stackdriver-nozzle/nozzle/envelope_test.go (hunk 1 above)
Comment (nrxus) [inline src/stackdriver-nozzle/nozzle/envelope_test.go]: Not sure - we made a story to figure out what to do with tags here: https://www.pivotaltracker.com/story/show/131571037

File: src/stackdriver-nozzle/config/config.go (hunk 4 above)
Comment (evandbrown) [inline src/stackdriver-nozzle/config/config.go]: I'd generally prefer flag parsing to happen inside main(). That makes sure anything that needs a flag/config value has to declare it in its method signature, and can't go grabbing initialized package-public vars.

File: src/stackdriver-nozzle/nozzle/envelope_test.go (hunk 1 above)
Comment (johnsonj) [inline src/stackdriver-nozzle/nozzle/envelope_test.go]: cool, thanks :)

File: src/stackdriver-nozzle/config/config.go (hunk 4 above)
Comment (nrxus) [inline src/stackdriver-nozzle/config/config.go]: I get where you're coming from. We did this to support the scripts, but I think I'm okay having the scripts just work off environment vars anyway.

**Comment by evandbrown** [comment]
Ahh, that makes sense. I'm fine adding a TODO on this one.

**Pull request merged**